docs/smart-contracts/modules.md

```diff
@@ -0,0 +1,564 @@
+The following example illustrates a module named **example** that is governed by the **admin-keyset** referenced in the first line of the module declaration:
+
+```pact
+;; Define and read keysets
+
+(module example "admin-keyset"
+    ;; module code goes here
+
+    ;; function calls
+)
+```
+
+The entire **example** module consists of the code within the opening and closing parentheses. 
+Top-level code, like `namespace` definitions and `keyset` definitions, are outside of these parentheses.
+For example, you enter a namespace and define a keyset for a module before starting the module declaration like this:
+
+```pact
+;; Enter a namespace
+(namespace "free")
+
+;; Define and read a keyset
+(define-keyset "free.admin-keyset" (read-keyset "admin-keyset"))
+
+(module example "admin-keyset"
+    ;; module code goes here
+
+    ;; function calls
+)
+```
+
+## Module governance
+
+With keyset governance like the previous example, the `admin-keyset` is defined outside of the module and checked and enforced at the module level. 
+Any attempt to upgrade the module, write to module tables, or access table functions directly requires the `admin-keyset` to sign the transaction.
+As an alternative to strict keyset enforcement, you can specify a _module governance_ capability in the module declaration to support a more generalized form of module governance. 
+By using a governance capability that references a `defcap` declaration in the module body, you can define more flexible models for enforcing access to Pact modules, tables, and functions. 
+
+For example, you can implement the same governance for the **example** module using a governance capability named **GOVERNANCE** like this:
+
+```pact
+(module example GOVERNANCE
+  ...
+  (defcap GOVERNANCE ()
+    (enforce-keyset "admin-keyset"))
+  ...
+)
+```
+
+Note that the capability name has no significance, except to indicate the purpose of the capability. 
+Its placement at the beginning of the module declaration is what identifies this capability as a module governance capability.
+
+It's worth noting that, when you initially deploy a module, the module governance capability is not invoked. 
+This behavior is different than when you use use a keyset.
+With a keyset, the keyset must always be defined and evaluated to ensure that the keyset exists before a module can be deployed.
+Therefore, you might consider module governance using a capability to be more risky than using a keyset. 
+You should test modules thoroughly when implementing module governance capabilities to ensure that you don't introduce bugs that might prevent a module from being upgraded.
+
+### Invoking governance
+
+Because the module governance capability is defined using the `defcap` keyword, its administrative function cannot be called directly. 
+It is automatically invoked in the following circumstances:
+
+- Module upgrade is attempted.
+- Module tables are directly accessed from outside of the module code.
```

Comment:
This is not correct

Reply:
I might have misunderstood what this is trying to communicate or maybe things have changed.
Where this came from:
https://docs.kadena.io/build/pact/advanced#generalized-module-governanceh-2066495538

### Governance capability and module admin

As a defcap, the governance function cannot be called directly by user code. It is automatically invoked in the following circumstances:

A module upgrade is being attempted
Module tables are being directly accessed outside the module code
A [module guard](https://docs.kadena.io/build/pact/advanced#module-guardsh1044818159) for this module is being enforced.
In these cases, the transaction is tested for elevated access to "module admin", defined as the grant of the module admin capability. This capability cannot be expressed in user code, so it cannot be installed, acquired, required or composed.

However, the implementing capability, here called GOVERNANCE, can be installed or acquired etc. If passed, this gets scoped like any normal capability, here over some protected code that only module admins can run.